pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go

```diff
@@ -640,23 +675,17 @@ func (sip *streamIngestionProcessor) flush() (*jobspb.ResolvedSpans, error) {
 		}
 	}
 
-	// Go through buffered checkpoint events, and put them on the channel to be
-	// emitted to the downstream frontier processor.
-	for partition, timestamp := range sip.bufferedCheckpoints {
-		// Each partition is represented by a span defined by the
-		// partition address.
-		spanStartKey := roachpb.Key(partition)
-		resolvedSpan := jobspb.ResolvedSpan{
-			Span:      roachpb.Span{Key: spanStartKey, EndKey: spanStartKey.Next()},
-			Timestamp: timestamp,
+	// Forward only resolved spans for each partition's spans rather than the
+	// entire frontier to avoid blowing up the size of the jobs table entry
+	for _, spec := range sip.spec.PartitionSpecs {
+		for _, span := range spec.Spans {
+			flushedCheckpoints.ResolvedSpans = append(flushedCheckpoints.ResolvedSpans, jobspb.ResolvedSpan{Span: span, Timestamp: sip.frontier.FrontierForSpans(spec.Spans...)})
```

Comment:
Do we want to do this at a greater resolution than just the spans we got for each partition from the topology?

Reply:
Save the whole frontier.

Reply:
Forwarded the whole frontier and added code in `TestTenantStreamingCheckpoint` to verify that the frontier contained all of the tables created for the source tenant 👍